pkg/reconciler/common/deployments.go

```diff
@@ -23,27 +23,28 @@ import (
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/errors"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
 	v1alpha1 "knative.dev/operator/pkg/apis/operator/v1alpha1"
-	kubeclient "knative.dev/pkg/client/injection/kube/client"
 )
 
 // CheckDeployments checks all deployments in the given manifest and updates the given
 // status with the status of the deployments.
 func CheckDeployments(ctx context.Context, manifest *mf.Manifest, instance v1alpha1.KComponent) error {
-	kube := kubeclient.Get(ctx)
 	status := instance.GetStatus()
+	status.MarkDeploymentsNotReady()
+	deployment := &appsv1.Deployment{}
```

Comment:
Does this need to move into the for loop now? I'd be anxious that it keeps "old" state from previously checked deployments otherwise, but not sure if that anxiety is justified.

Reply:
`Scheme.Convert` would be pretty broken if that can happen, but I can create it in the loop easy enough.

Reply:
Done